### ISSUE TITLE
New Spark connection shows error instead of the form fields

### ISSUE THREAD

**Issue by ronblum** [rstudio/rstudio#13296]
### System details

    RStudio Edition : Desktop and Server
    RStudio Version : 2022.07.2+576, 2023.06.0+421.pro1, 2023.06.1+438, 2023.08.0-daily+60
    OS Version      : MacOS 13.4 (desktop) and Ubuntu 20.04.5 (server)
    R Version       : 3.6.3 and 4.3.0

### Steps to reproduce the problem

In the Connections Pane: <kbd>New Connection -> Spark</kbd>

### Describe the problem in detail

A "subscript out of bounds" error appears instead of the form fields. The error is also in the Console Pane.

<img width="528" alt="spark" src="https://github.com/rstudio/rstudio/assets/31009952/40a5ae4c-a9f2-4098-a8e8-eab970a150fd">

>Listening on http://127.0.0.1:6870
Warning: Error in [[: subscript out of bounds
  76: <Anonymous>
  75: FUN
  74: lapply
  73: spark_versions
  72: spark_default_version
  61: ui [/Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/sparklyr/rstudio/shinycon/app.R#121]
   2: shiny::runApp
   1: shiny::runGadget

### Describe the behavior you expected

The Spark setup UI appears.

Tested with `sparklyr` 1.8.1 and `shiny` 1.7.4.

- [ ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [ ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [ ] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
Is this potentially an issue in `sparklyr` itself? Or does this work correctly with older releases of RStudio as well?

**Comment by kevinushey** [comment]
It seems like a `sparklyr` issue to me:

```
> sparklyr::spark_versions()
Error in validVersions[with(validVersions, order(hadoop, decreasing = TRUE)),  : 
  subscript out of bounds
```

**Comment by kevinushey** [comment]
It seems like this might be resolved in the development version of sparklyr? I've filed https://github.com/sparklyr/sparklyr/issues/3350.

**Comment by ronblum** [comment]
Yep! It works correctly with the dev version. Thanks!

I'm going to leave this open for now, to retest after the new version of sparklyr is live.

**Comment by edgararuiz** [comment]
The new versions of `sparklyr` is now available in CRAN (v 1.8.2)

**Comment by kevinushey** [comment]
I can confirm things appear to be fixed -- thanks for taking a look so quickly @edgararuiz!
